### PR TITLE
AUD-764 Add ChallengeReward desktop notification

### PR DESCRIPTION
**Pull request by Kyle-Shanks**
### Description

Add the NotificationType, notification block component, and formatText code for ChallengeReward notifications

### How Has This Been Tested?

manually

### CODE REVIEW

File: src/containers/notification/NotificationPanel.tsx [hunk 1]
```diff
@@ -33,7 +33,7 @@ type OwnProps = {
   isElectron: boolean
   status: Status
   hasMore: boolean
-  notifications: any
+  notifications: Notification[]
```
Comment (jowlee) [inline src/containers/notification/NotificationPanel.tsx:36]: 🔥 🙏

File: src/containers/notification/components/TwitterShare.tsx [hunk 2]
```diff
@@ -149,6 +150,12 @@ export const getRemixCosignText = async (
     link
   }
 }
+
+export const getRewardsText = (notification: ChallengeReward) => ({
+  text: `I earned $AUDIO for completing challenges on @AudiusProject #AudioRewards`,
+  link: ''
```
Comment (jowlee) [inline src/containers/notification/components/TwitterShare.tsx:156]: link?

**Review by jowlee — APPROVED** [review]

File: src/containers/notification/components/TwitterShare.tsx (hunk 2 above)
Comment (jowlee) [reply src/containers/notification/components/TwitterShare.tsx:156]: Maybe make the link optional and not require in the `openTwitterLink` func
ref: https://github.com/AudiusProject/audius-client/blob/be456a65c14b2a2c6e85b734c2b258f0f122eb37/src/utils/tweet.ts#L1

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/kj-AUD-764-Add-challenge-reward-desktop-notification

File: src/containers/notification/components/TwitterShare.tsx (hunk 2 above)
Comment (Kyle-Shanks) [reply src/containers/notification/components/TwitterShare.tsx:156]: Not sure if there is supposed to be a link for this, could be the profile page I guess? This is placeholder while I check with Julian

**Review by raymondjacobson — APPROVED** [review]
nice! let's also figure out what the copy should be for badges notification. looks great

**Pull request merged**